.gitignore

```diff
@@ -27,3 +27,6 @@ stravalib/docs/api
 .vscode/
 stravalib-2023/
 .nox/*
+
+# Ignore Sphinx auto-generated API stubs #
+docs/reference/api
```

Comment:
this is really the only change in ths PR. the other 70 files are api stubs that i'm removing from versioning given sphinx will always rebuild them. this will help us maintain the api docs in a cleaner format.